compiler/haskell-ide-core/src/Development/IDE/Functions/AtPoint.hs

```diff
@@ -74,10 +86,36 @@ atPoint tcs srcSpans pos = do
         Just name -> any (`isInfixOf` show name) ["==", "showsPrec"]
         Nothing -> False
 
-locationsAtPoint :: Position -> [SpanInfo] -> [Location]
-locationsAtPoint pos = map srcSpanToLocation
-                     . mapMaybe (SpanInfo.getSrcSpan . spaninfoSource)
-                     . spansAtPoint pos
+locationsAtPoint :: IdeOptions -> PackageDynFlags -> Position -> [SpanInfo] -> Action [Location]
+locationsAtPoint IdeOptions{..} pkgState pos =
+    fmap (map srcSpanToLocation) .
+    mapMaybeM (getSpan . spaninfoSource) .
+    spansAtPoint pos
+  where getSpan :: SpanSource -> Action (Maybe SrcSpan)
+        getSpan NoSource = pure Nothing
+        getSpan (Span sp) = pure $ Just sp
+        getSpan (Named name) = case nameSrcSpan name of
+            sp@(RealSrcSpan _) -> pure $ Just sp
+            UnhelpfulSpan _ -> runMaybeT $ do
+                -- This case usually arises when the definition is in an external package.
+                -- In this case the interface files contain garbage source spans
+                -- so we instead read the .hie files to get useful source spans.
+                let mod = nameModule name
+                let unitId = moduleUnitId mod
+                pkgConfig <- MaybeT $ pure $ lookupPackageConfig unitId pkgState
+                hiePath <- MaybeT $ liftIO $ optLocateHieFile pkgConfig mod
+                hieFile <- MaybeT $ use (GetHieFile hiePath) ""
+                avail <- MaybeT $ pure $ listToMaybe (filterAvails (eqName name) $ hie_exports hieFile)
+                srcPath <- MaybeT $ liftIO $ optLocateSrcFile pkgConfig mod
+                -- The location will point to the source file used during compilation.
+                -- This file might no longer exists and even if it does the path will be relative
+                -- to the compilation directory which we don’t know.
+                let span = setFileName srcPath $ nameSrcSpan $ availName avail
+                pure span
+        eqName :: Name -> Name -> Bool
+        eqName n n' = nameOccName n == nameOccName n' && nameModule n == nameModule n'
```

Comment:
These already have an Eq on them. I assume the unique counts might be different? If so, could we get a comment to that effect. Perhaps also put this in UtilGHC?

Reply:
Uniques and source spans, I’ve added a comment. I’ll move it to `UtiGHC`once we end up using it somewhere else.